### PR TITLE
TEC-658, add blog identities

### PR DESCRIPTION
**Pull request by gcedo**
Lives @ http://gcedo.github.io/component-header/

### CODE REVIEW

**Comment by keithamus** [comment]
Oh my.

**Comment by gcedo** [comment]
Something wrong?

**Comment by keithamus** [comment]
I would say it is a good idea to move all of the svgs into a separate folder, perhaps `banners`, and then have a `build:svg` task which crushes the svgs down. We do this with [component-icon](https://github.com/economist-components/component-icon) (see the [icons folder](https://github.com/economist-components/component-icon/tree/master/icons), the [build:icons script](https://github.com/economist-components/component-icon/blob/master/package.json#L40) and the [svgo.yml](https://github.com/economist-components/component-icon/blob/master/.svgo.yml)).

**Comment by keithamus** [comment]
What would have been really cool is to only have 1 svg for desktop and mobile, and use media queries inside the SVG CSS to switch between them. As they have both the same imagery and colours, we could really hugely optimise on how they work.

File: src/blogCustomisation.css [hunk 1]
```diff
@@ -31,19 +45,88 @@
...
+/* buttonwood */
+
+.header--blog21003991 {
+  background-color: #bfe7f1;
+}
+
+.header--blog21003991 .header__title,
+.header--blog21003991 .header__title-link,
+.header--blog21003991 .header__text {
+  color: var(--color-beijing, #121212);
```
Comment (keithamus) [inline src/blogCustomisation.css]: I'm pretty sure we discussed this - we should strip out all of the economist colour palette vars and go with overridable vars for this component.

File: src/blogCustomisation.css [hunk 2]
```diff
@@ -31,19 +45,88 @@
...
+/* erasmus */
+
+.header--blog21572045 {
+  background-color: #bd902d;
+}
+
+.header--blog21572045 .header__title,
+.header--blog21572045 .header__title-link,
+.header--blog21572045 .header__text {
+  color: var(--color-beijing, #121212);
```
Comment (keithamus) [inline src/blogCustomisation.css]: Again here.

**Comment by gcedo** [comment]
I have never done that, but if you point me to an example I can do it.

File: src/blogCustomisation.css [hunk 3]
```diff
@@ -31,19 +45,88 @@
...
+/* gulliver */
+
+.header--blog21003986 {
+  background-color: #5ac5c4;
+}
+
+.header--blog21003986 .header__title,
+.header--blog21003986 .header__title-link,
+.header--blog21003986 .header__text {
+  color: var(--color-beijing, #121212);
```
Comment (keithamus) [inline src/blogCustomisation.css]: And here.

File: src/blogCustomisation.css [hunk 4]
```diff
@@ -31,19 +45,88 @@
...
+/* prospero */
+
+.header--blog21008659 {
+  background-color: #ece8e1;
+}
+
+.header--blog21008659 .header__title,
+.header--blog21008659 .header__title-link,
+.header--blog21008659 .header__text {
+  color: var(--color-beijing, #121212);
```
Comment (keithamus) [inline src/blogCustomisation.css]: And here.

File: src/example.js [hunk 5]
```diff
@@ -7,22 +7,79 @@ if (typeof location !== 'undefined' && /github.io\//.test(String(location))) {
...
-      itemProp="blog"
-      link={{
-        href: 'http://www.google.com',
-        title: 'Clicke here and go somewhere',
-      }}
-    />
+    {
+      blogs.map((blog, index) =>
+        <Header
+          className={`header--blog header--blog${ blog.id }`}
```
Comment (keithamus) [inline src/example.js]: It'd be much cleaner to use `blog.slug` here. It would make the CSS tolerable.

**Comment by keithamus** [comment]
So, you already have CSS specific info about each blog identity (because you set the background color for each one). It would not be too difficult to rework the SVGs to just include the base imagery, and then in the CSS set things like `height` & `background-position` to recreate the two sizes for desktop and mobile. It would not be too difficult to also make these much more fluid. For example:

```css
/* This is just an example of how this could work, these values are pulled out of thin air */
.header--blog-bagehot {
  background-color: #b51915;
  background-image: url('assets/header-bagehot.svg') no-repeat top center;
  height: 600px;

  @media (max-width: 600px) {
    background-position: right middle;
    height: 200px;
  }
}
```

**Comment by gcedo** [comment]
Ok, I am working on creating the sprite. For the second bit, we need to ask the designers to redo all of the assets I guess, given, for instance http://gcedo.github.io/component-header/assets/header-democracyinamerica.svg. Would that be correct.

**Pull request closed without merging**